tests/ctl/conftest.py

```diff
@@ -8,33 +8,56 @@
 import json
 import os
 from datetime import datetime
-from typing import Any, AsyncGenerator, Dict, Generator, Union
-from uuid import uuid4
+from typing import Dict, Generator, Union
 
 import pytest
+import requests
 import yaml
+from _pytest.monkeypatch import MonkeyPatch
 from fideslang import models
 from fideslib.cryptography.schemas.jwt import (
     JWE_ISSUED_AT,
     JWE_PAYLOAD_CLIENT_ID,
     JWE_PAYLOAD_SCOPES,
 )
-from fideslib.models.client import ClientDetail
 from fideslib.oauth.jwt import generate_jwe
-from fideslib.oauth.scopes import PRIVACY_REQUEST_READ, SCOPES
-from sqlalchemy.orm import Session
-from sqlalchemy.orm.exc import ObjectDeletedError
+from sqlalchemy.ext.asyncio import AsyncSession, create_async_engine
+from sqlalchemy.orm import sessionmaker
 from starlette.testclient import TestClient
 
 from fides.api import main
-from fides.api.ctl.database.session import engine, sync_session
-from fides.api.ctl.sql_models import FidesUser, FidesUserPermissions
+from fides.api.ctl.sql_models import FidesUser
 from fides.ctl.core import api
 from fides.ctl.core.config import FidesConfig, get_config
 
 TEST_CONFIG_PATH = "tests/ctl/test_config.toml"
 TEST_INVALID_CONFIG_PATH = "tests/ctl/test_invalid_config.toml"
 TEST_DEPRECATED_CONFIG_PATH = "tests/ctl/test_deprecated_config.toml"
+CONFIG = get_config()
+
+
+orig_requests_get = requests.get
+orig_requests_post = requests.post
+orig_requests_put = requests.put
+orig_requests_patch = requests.patch
+orig_requests_delete = requests.delete
+
+
+@pytest.fixture(scope="session")
+def monkeysession():
+    """monkeypatch fixture at the session level instead of the function level"""
+    mpatch = MonkeyPatch()
+    yield mpatch
+    mpatch.undo()
+
+
+@pytest.fixture(autouse=True, scope="session")
+def monkeypatch_requests(test_client, monkeysession) -> None:
+    monkeysession.setattr(requests, "get", test_client.get)
+    monkeysession.setattr(requests, "post", test_client.post)
+    monkeysession.setattr(requests, "put", test_client.put)
+    monkeysession.setattr(requests, "patch", test_client.patch)
+    monkeysession.setattr(requests, "delete", test_client.delete)
```

Comment:
@ThomasLaPiana your monkeypatching suggestion worked very well! My main lingering concern would be that I don't want to be too aggressive with this monkeypatching - so I'm just doing this on the ctl-side. The ops side uses the `requests` library a lot to talk to connectors.

Reply:
@pattisdr that is correct, it monkeypatches _everywhere_ (due to how Python works under the hood I guess) so you then need to explicitly un-patch it, but this model works fine for our current use!